tests/acceptance/features/bootstrap/WebDav.php

```diff
@@ -1184,7 +1184,13 @@ public function contentOfFileShouldBePlusEndOfLine(string $fileName, string $con
 	public function contentOfFileForUserShouldBe(string $fileName, string $user, string $content):void {
 		$user = $this->getActualUsername($user);
 		$this->downloadFileAsUserUsingPassword($user, $fileName);
-		$this->downloadedContentShouldBe($content);
+		$actualStatus = $this->response->getStatusCode();
+		if ($actualStatus !== 200) {
+			throw new Exception(
+				"Expected status code to be '200', but got '$actualStatus'"
+			);
+		}
```

Comment:
This is not wrong but I feel like there should be a status code check in the scenario step itself
```
Then the htttp status code should be <expected status code>
```
Does the scenario that's intermittently failing have the status code check? Maybe add there instead of here

Reply:
Ah okay the scenario is like this
```
 Scenario Outline: uploading a file with checksum should work for file with special characters                                                              # /drone/src/tests/acceptance/features/coreApiMain/checksums.feature:334
    Given using <dav_version> DAV path                                                                                                                       # FeatureContext::usingOldOrNewDavPath()
    When user "Alice" uploads file "filesForUpload/textfile.txt" to <renamed_file> with checksum "MD5:d70b40f177b14b470d1756a3c12b963a" using the WebDAV API # ChecksumContext::userUploadsFileToWithChecksumUsingTheAPI()
    Then the HTTP status code should be "201"                                                                                                                # FeatureContext::thenTheHTTPStatusCodeShouldBe()
    And the content of file <renamed_file> for user "Alice" should be:                                                                                       # FeatureContext::contentOfFileForUserShouldBePyString()
      """
      This is a testfile.
      
      Cheers.
      """
```

Reply:
There isn't any status code check when file is downloaded. This part of the code is in the step where the content of the file is checked.
Related Scenario:
https://github.com/owncloud/ocis/blob/563bf34870fec2c544bb00e0ee03133f505d5ea9/tests/acceptance/features/coreApiMain/checksums.feature#L334

Reply:
Yes we cannot check the status code in steps for this case